src/components/input/InputGroup.js

```diff
@@ -0,0 +1,49 @@
+import classnames from 'classnames';
+
+import { downcastRef } from '../../util/typing';
+
+/**
+ * @typedef {import('../../types').PresentationalProps} CommonProps
+ * @typedef {import('preact').JSX.HTMLAttributes<HTMLElement>} HTMLAttributes
+ */
+
+// These styles may be applied to input components to adapt them when in
+// an InputGroup
+export const inputGroupStyles = classnames(
+  // All inputs within an InputGroup should have a border. Turn off border-radius
+  'input-group:border input-group:rounded-none',
+  // Restore border-radius on the leftmost and rightmost components in the group
+  'input-group:first:rounded-l-sm input-group:last:rounded-r-sm',
+  // "Collapse" borders between input components
+  'input-group:border-l-0 input-group:first:border-l'
+);
```

Comment:
This was a new pattern I ran into: the desire to reuse a set of classes, almost like a mixin, that didn't stand on their own (as a component)...open to other thoughts on how to organize this.